caddy: Allow adding external plugins (#259275)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2311.section.md

  If you use this feature, updates to CoreDNS may require updating `vendorHash` by following these steps again.

- Caddy can now be built with external plugins by overriding `externalPlugins` and `vendorHash` arguments like this:

  ```
  services.caddy = {
    enable = true;
    package = pkgs.caddy.override {
      externalPlugins = [

>         {name = "coraza"; repo = "github.com/corazawaf/coraza-caddy/v2"; version = "v2.0.0-rc.3";}
>         {name = "cache-handler"; repo = "github.com/caddyserver/cache-handler"; version = "v0.9.0";}
>         {name = "ratelimit"; repo = "github.com/mholt/caddy-ratelimit"; version = "2dc0d586f0b87e983757c403bc0929ddeb84a537";}

Reviewer: ```suggestion
        { name = "coraza"; repo = "github.com/corazawaf/coraza-caddy/v2"; version = "v2.0.0-rc.3"; }
        { name = "cache-handler"; repo = "github.com/caddyserver/cache-handler"; version = "v0.9.0"; }
        { name = "ratelimit"; repo = "github.com/mholt/caddy-ratelimit"; version = "2dc0d586f0b87e983757c403bc0929ddeb84a537"; }
```

      ];
      vendorHash = "<SRI hash>";
    };
  };
  ```

>   To get the necessary SRI hash, set `vendorHash = "";`. The build will fail and produce the correct `vendorHash` in the error message.

Reviewer: ```suggestion
  To get the necessary SRI hash for the vendor directory, first set `vendorHash = "";`. Than build the next build will fail and log the correct `vendorHash` in the error message.
```

>   If you use this feature, updates to Caddy may require updating `vendorHash` by following these steps again.

Reviewer: ```suggestion
  If you use this feature, updates to Caddy likely require updating `vendorHash` by following these steps again.
```
